### PR TITLE
Add/feature module

### PR DESCRIPTION
**Pull request by circlecube**
## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: bootstrap.php [hunk 1]
```diff
@@ -1,78 +1,7 @@
...
-use NewfoldLabs\WP\Module\Performance\CacheTypes\Skip404;
-use NewfoldLabs\WP\Module\Performance\Performance;
-use NewfoldLabs\WP\Module\Performance\ResponseHeaderManager;
-use NewfoldLabs\WP\ModuleLoader\Container;
+namespace NewfoldLabs\WP\Module\Performance;
 
-use function NewfoldLabs\WP\Module\Performance\getCacheLevel;
-use function NewfoldLabs\WP\ModuleLoader\register;
-use function NewfoldLabs\WP\Context\getContext;
+require_once __DIR__ . '/includes/PerformanceFeature.php';
```
Comment (wpscholar) [inline bootstrap.php:5]: We should use a Composer file autoloader for this.

File: includes/PerformanceFeature.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+							Skip404::onDeactivation();
+							File::onDeactivation();
+							Browser::onDeactivation();
+							// Remove all headers from .htaccess
+							$responseHeaderManager = new ResponseHeaderManager();
+							$responseHeaderManager->removeAllHeaders();
+						}
+					);
+				}
+			);
```
Comment (wpscholar) [inline includes/PerformanceFeature.php:79]: As discussed, I don't think we'll want to conditionally run activation/deactivation hooks. I also don't think that these would run early enough if they are registered in an action hook callback.

File: bootstrap.php (hunk 1 above)
Comment (circlecube) [reply bootstrap.php]: Got it

File: includes/PerformanceFeature.php (hunk 2 above)
Comment (circlecube) [reply includes/PerformanceFeature.php]: I've got these hooks integrated better and moved into the PerformanceFeatureHooks class.

File: includes/PerformanceFeatureHooks.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	 */
+	public function pluginHooks( Container $container ) {
+		register_activation_hook( $container->plugin()->file, array( $this, 'onActivation' ) );
+		register_deactivation_hook( $container->plugin()->file, array( $this, 'onDeactivation' ) );
+	}
+
+	/**
+	 * Activation hook to perform when plugin is activated or feature is enabled
+	 */
+	public function onActivation() {
```
Comment (wpscholar) [inline includes/PerformanceFeatureHooks.php:42]: @circlecube Looks good, but let's also perform the `onActivation` when the module is turned on and the `onDeactivation` when the module is turned off.

**Review by wpscholar — CHANGES_REQUESTED** [review]

**Review by wpscholar — APPROVED** [review]

**Review by wpscholar — APPROVED** [review]

**Pull request merged**